Sources/Private/Experimental/Layers/BaseCompositionLayer.swift

```diff
@@ -62,18 +64,18 @@ class BaseCompositionLayer: CALayer, AnimationLayer {
   /// and all child `AnimationLayer`s.
   ///  - Can be overridden by subclasses, which much call `super`.
   func setupAnimations(context: LayerAnimationContext) {
-    addVisibilityAnimation(
-      inFrame: CGFloat(baseLayerModel.inFrame),
-      outFrame: CGFloat(baseLayerModel.outFrame),
-      context: context)
-
     transformLayer.addTransformAnimations(for: baseLayerModel.transform, context: context)
 
-    // Add the opacity animation to _only_ the `contentsLayer`
-    //  - The opacity should affect the content rendered directly by this layer,
+    // Add the opacity and visibility animations to _only_ the `contentsLayer`
+    //  - These animations should affect the content rendered directly by this layer,
     //    but _not_ any child `LayerModel`s.
     contentsLayer.addOpacityAnimation(from: baseLayerModel.transform, context: context)
 
+    contentsLayer.addVisibilityAnimation(
```

Comment:
This change fixes an issue where layers could disappear unexpectedly (because one of their parents were hidden)

<img width="870" alt="image" src="https://user-images.githubusercontent.com/1811727/148592043-7da15e23-3f36-4190-a41b-a6fa1956eebe.png">